src/realm/table.cpp

```diff
@@ -422,8 +422,13 @@ TableKey Table::get_key_direct(Allocator& alloc, ref_type top_ref)
 }
 
 
-void Table::init(ref_type top_ref, ArrayParent* parent, size_t ndx_in_parent, bool is_writable)
+void Table::init(ref_type top_ref, ArrayParent* parent, size_t ndx_in_parent, bool is_writable, bool is_frozen)
 {
+    if (is_writable)
```

Comment:
I would prefer to have this written as one
`REALM_ASSERT((!is_writable || is_frosen) && (!is_frozen || is_writable));`

Reply:
but that would be wrong, right?